### PR TITLE
[t242] - popcornOptions & generatePopcornString

### PR DESCRIPTION
**Pull request by secretrobotron**
1. Verified the passthrough of popcornOptions to string generator.
2. Added accessor to change popcornOptions if desired.
3. Added tests.

### CODE REVIEW

File: src/core/popcorn-wrapper.js [hunk 1]
```diff
@@ -152,55 +231,25 @@ define( [ "core/logger", "core/eventmanager" ], function( Logger, EventManager )
...
-      // call certain player function depending on the regexResult
-      popcornString += constructPlayer( _mediaType );
+      // special case for basePlayer, since it doesn't require as much of a harness
+      if( _mediaType === "baseplayer" ) {
+        popcornString +=  "Popcorn.player( 'baseplayer' );\n" +
+                          "var popcorn = Popcorn.baseplayer( '#" + target + "', " + popcornOptions + " );\n";
+      }
+      else{
+        // just try to use Popcorn.smart to detect/setup video
+        popcornString += "var popcorn = Popcorn.smart( '#" + target + "', '" + url + "'" + popcornOptions + " );\n";
```
Comment (cadecairos) [inline src/core/popcorn-wrapper.js:246]: Looks like a comma's missing here before " + popcornOptions"

File: test/core.js [hunk 2]
```diff
@@ -538,7 +538,7 @@
 
     expect( expected );
     Butter({
-      config: "../config/default.conf",
+      config: "../config/test.conf",
```
Comment (cadecairos) [inline test/core.js:541]: Was this supposed to be still using test.conf?

Comment (secretrobotron) [inline test/core.js:541]: Yes. Tests are to use test.conf to disable the UI.

File: src/core/popcorn-wrapper.js (hunk 1 above)
Comment (secretrobotron) [inline src/core/popcorn-wrapper.js:246]: It's not. It gets placed in the code above when popcornOptions is created: https://github.com/mozilla/butter/pull/129/files#L3R221

Comment (cadecairos) [inline src/core/popcorn-wrapper.js:246]: haha thats sneaky.

**Pull request merged**